lgc/patch/FragColorExport.cpp

```diff
@@ -130,56 +140,54 @@ Value *FragColorExport::handleColorExportInstructions(Value *output, unsigned hw
     }
   }
 
-  const auto undefFloat = PoisonValue::get(builder.getFloatTy());
-  const auto undefFloat16x2 = PoisonValue::get(FixedVectorType::get(builder.getHalfTy(), 2));
-
   switch (expFmt) {
-  case EXP_FORMAT_32_R: {
-    compCount = 1;
-    comps[0] = convertToFloat(comps[0], signedness, builder);
-    break;
-  }
-  case EXP_FORMAT_32_GR: {
-    if (compCount >= 2) {
+  case EXP_FORMAT_32_R:
+  case EXP_FORMAT_32_GR:
+  case EXP_FORMAT_32_ABGR: {
+    if (expFmt == EXP_FORMAT_32_GR && compCount >= 2)
       compCount = 2;
-      comps[0] = convertToFloat(comps[0], signedness, builder);
-      comps[1] = convertToFloat(comps[1], signedness, builder);
-    } else {
+    else if (expFmt != EXP_FORMAT_32_ABGR)
       compCount = 1;
-      comps[0] = convertToFloat(comps[0], signedness, builder);
+
+    for (unsigned idx = 0; idx < compCount; ++idx) {
+      unsigned compMask = 1 << idx;
+      if (compMask & channelWriteMask) {
+        exports[idx] = convertToFloat(comps[idx], signedness, builder);
+        exportMask |= compMask;
+      }
     }
     break;
   }
   case EXP_FORMAT_32_AR: {
-    if (compCount == 4) {
+    if (compCount == 4)
       compCount = 2;
-      comps[0] = convertToFloat(comps[0], signedness, builder);
-      comps[1] = convertToFloat(comps[3], signedness, builder);
-    } else {
+    else
       compCount = 1;
-      comps[0] = convertToFloat(comps[0], signedness, builder);
+    for (unsigned idx = 0; idx < compCount; ++idx) {
+      unsigned j = (idx == 1) ? 3 : idx;
+      unsigned compMask = 1 << j;
+      if (compMask & channelWriteMask) {
+        exports[idx] = convertToFloat(comps[j], signedness, builder);
+        exportMask |= compMask;
+      }
```

Comment:
I think this would be cleaner without the loop: set exports[0], then if compCount is 4, set exports[1] using a simple if.

Reply:
Fixed.

Reply:
Did you push the new version? I don't see an update.

Reply:
Oops. I forgot to commit the new version. I have updated in the latest push. Please review.